packages/babel-parser/src/parser/util.js

```diff
@@ -268,4 +268,26 @@ export default class UtilParser extends Tokenizer {
       throw error;
     }
   }
+
+  checkExpressionErrors(
+    refExpressionErrors: ?ExpressionErrors,
+    andThrow: boolean,
+  ) {
+    if (!refExpressionErrors) return false;
+    const { shorthandAssign } = refExpressionErrors;
+    if (!andThrow) return shorthandAssign >= 0;
+    if (shorthandAssign >= 0) {
+      this.unexpected(shorthandAssign);
```

Comment:
(In a separate PR)
Probably this error would be easily recoverable.

Reply:
Yeah, I try to preserve the error output of shorthand assign error so it's indeed a refactor.